CRM/Custom/Form/CustomDataTrait.php

```diff
@@ -183,4 +183,24 @@ protected function getSubmittedCustomFields(): array {
     return $fields;
   }
 
+  /**
+   * Get the submitted custom fields in Api4 format.
+   *
+   * @return array
+   */
+  protected function getSubmittedCustomFieldsForApi4(): array {
+    $fields = [];
+    foreach ($this->getSubmittedValues() as $label => $field) {
+      if ($id = CRM_Core_BAO_CustomField::getKeyID($label)) {
+        $cf = \Civi\Api4\CustomField::get(FALSE)
+          ->addSelect('custom_group_id:name', 'name')
+          ->addWhere('id', '=', $id)
+          ->execute()
+          ->single();
+        $fields[$cf['custom_group_id:name'] . '.' . $cf['name']] = $field;
+      }
+    }
+    return $fields;
```

Comment:
Check out the inverse function `CRM_Core_BAO_CustomField::getLongNameFromShortName()`